src/core/sys/darwin/mach/getsect.d

```diff
@@ -22,12 +351,121 @@ else version (WatchOS)
     version = Darwin;
 
 version (Darwin):
-extern (C):
-nothrow:
-@nogc:
 
 public import core.sys.darwin.mach.loader;
 
-const(section)*    getsectbynamefromheader(in mach_header* mhp, in char* segname, in char* sectname);
-const(section_64)* getsectbynamefromheader_64(in mach_header_64* mhp, in char* segname, in char* sectname);
+import core.stdc.config : c_ulong;
+
+char* getsectdata(
+    in char* segname,
+    in char* sectname,
+    c_ulong *size
+);
+
+char* getsectdatafromFramework(
+    in char* FrameworkName,
+    in char* segname,
+    in char* sectname,
+    c_ulong* size
+);
+
+c_ulong get_end();
+c_ulong get_etext();
+c_ulong get_edata();
+
+// Runtime interfaces for 64-bit Mach-O programs.
+version (D_LP64)
+{
+    const(section_64)* getsectbyname(
+        in char* segname,
+        in char* sectname
+    );
+
+    ubyte* getsectiondata(
+        in mach_header_64* mhp,
+        in char* segname,
+        in char* sectname,
+        c_ulong* size
+    );
+
+    const(segment_command_64)* getsegbyname(
+        in char* segname
+    );
+
+    ubyte* getsegmentdata(
+        in mach_header_64* mhp,
+        in char* segname,
+        c_ulong* size
+    );
+}
+
+// Runtime interfaces for 32-bit Mach-O programs.
+else
+{
+    const(section)* getsectbyname(
+        in char* segname,
+        in char* sectname
+    );
+
+    ubyte* getsectiondata(
+        in mach_header* mhp,
+        in char* segname,
+        in char* sectname,
+        c_ulong* size
+    );
+
+    const(segment_command)* getsegbyname(
+        in char* segname
+    );
+
+    ubyte* getsegmentdata(
+        in mach_header* mhp,
+        in char* segname,
+        c_ulong* size
+    );
+}
+
+// Interfaces for tools working with 32-bit Mach-O files.
+
+ubyte* getsectdatafromheader(
+    in mach_header* mhp,
+    in char* segname,
+    in char* sectname,
+    c_ulong* size
+);
+
+const(section)* getsectbynamefromheader(
+    in mach_header* mhp,
+    in char* segname,
+    in char* sectname
+);
+
+const(section)* getsectbynamefromheaderwithswap_64(
```

Comment:
@jacob-carlborg: I take it the `_64` suffix is a typo? LDC doesn't like `extern(C)` overloads.

Reply:
Yes, that's a typo.

Reply:
https://github.com/dlang/druntime/pull/2229